modules/pipeline/providers/resourcegc/compensator.go

```diff
@@ -82,27 +80,39 @@ func (r *Reconciler) getNeedGCPipelines(pageNum int, isSnippet bool) ([]spec.Pip
 	}
 	req.AllSources = true
 	req.IncludeSnippet = isSnippet
-	pipelines, _, _, _, err := r.dbClient.PageListPipelines(req)
+	pipelines, _, _, _, err := p.dbClient.PageListPipelines(req)
 	if err != nil {
 		return nil, 0, fmt.Errorf("failed to compensate pipeline req %v err: %v", req, err)
 	} else {
-		for _, p := range pipelines {
-			if !p.Status.IsEndStatus() {
+		for _, pt := range pipelines {
```

Comment:
strange `pt`.
I think you can use `r` as receiver name (abbr for `resourcegc`).

Reply:
adjusted